src/engine/Settings.cs

```diff
@@ -64,74 +116,158 @@ private Settings()
 
     /// <summary>
     ///   Number of quick saves to keep
     /// </summary>
     public int MaxQuickSaves { get; set; } = 5;
 
     /// <summary>
-    ///   Number of auto saves to keep
+    ///   When true cheats are enabled
     /// </summary>
-    public int MaxAutoSaves { get; set; } = 5;
+    public bool CheatsEnabled { get; set; } = false;
 
     /// <summary>
-    ///   This can be freely adjusted to adjust the performance The
-    ///   higher this value is the smaller the size of the simulated
-    ///   cloud is and the performance is better.
+    ///   If true an auto-evo run is started during gameplay,
+    ///   taking up one of the background threads.
     /// </summary>
-    public int CloudResolution { get; set; } = 2;
+    public bool RunAutoEvoDuringGamePlay { get; set; } = true;
 
-    /// <summary>
-    ///   If this is over 0 then this limits how often compound clouds
-    ///   are updated. The default value of 0.020 at 60 FPS makes
-    ///   every other frame not update the clouds.
-    /// </summary>
-    /// <remarks>
-    ///   <para>
-    ///     This should be made user configurable for different
-    ///     computers. The choises should probably be:
-    ///     0.0f, 0.020f, 0.040f, 0.1f, 0.25f
-    ///   </para>
-    /// </remarks>
-    public float CloudUpdateInterval { get; set; } = 0.040f;
+    public int CloudSimulationWidth => Constants.CLOUD_X_EXTENT / CloudResolution;
+
+    public int CloudSimulationHeight => Constants.CLOUD_Y_EXTENT / CloudResolution;
+
+    public static bool operator ==(Settings lhs, Settings rhs)
+    {
+        return Equals(lhs, rhs);
+    }
+
+    public static bool operator !=(Settings lhs, Settings rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    public override bool Equals(object obj)
+    {
+        if (obj == null)
+        {
+            return false;
+        }
+
+        if (GetType() != obj.GetType())
+        {
+            return false;
+        }
+
+        return Equals((Settings)obj);
+    }
+
+    public bool Equals(Settings obj)
+    {
+        // Compare all properties in the two objects for equality.
+        var type = GetType();
+
+        var properties = type.GetProperties();
+
+        foreach (var prop in properties)
+        {
+            // Returns if any of the properties don't match.
+            object thisValue = type.GetProperty(prop.Name)?.GetValue(this, null);
+            object objValue = type.GetProperty(prop.Name)?.GetValue(obj, null);
+
+            if (thisValue != objValue && (thisValue == null || !thisValue.Equals(objValue)))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public override int GetHashCode()
+    {
+        int hashCode = 17;
+
+        var type = GetType();
+
+        var properties = type.GetProperties();
+
+        foreach (var prop in properties)
+        {
+            int? nextHash = type.GetProperty(prop.Name)?.GetHashCode();
```

Comment:
Same thing as my previous comment.

Reply:
By previous comment I meant this: https://github.com/Revolutionary-Games/Thrive/pull/1588#discussion_r485408560
I wrote this before adding the comment that is between these.